src/pybind/mgr/dashboard/services/nvmeof_conf.py

```diff
@@ -112,3 +136,45 @@ def from_cert_store(cls, entity: str, service_name: str, key=False):
             # just return None if any orchestrator error is raised
             # otherwise nvmeof api will raise this error and doesn't proceed.
             return None
+
+    @classmethod
+    def _get_name_url_for_group(cls, gateways, group):
+        try:
+            orch = OrchClient.instance()
+            for service_name, svc_config in gateways.items():
+                # get the group name of the service and match it against the
+                # group name provided
+                group_name_from_svc = orch.services.get(service_name)[0].spec.group
+                if group == group_name_from_svc:
+                    running_daemons = cls._get_running_daemons(orch, service_name)
+                    config = cls._get_running_daemon_svc_config(svc_config, running_daemons)
+
+                    if config:
+                        return service_name, config['service_url']
+            return None
+
+        except OrchestratorError:
+            return cls._get_default_service(gateways)
+
+    @classmethod
+    def _get_running_daemons(cls, orch, service_name):
+        # get the running nvmeof daemons
+        daemons = [d.to_dict()
+                   for d in orch.services.list_daemons(service_name=service_name)]
+        return [d['daemon_name'] for d in daemons
+                if d['status_desc'] == 'running']
+
+    @classmethod
+    def _get_running_daemon_svc_config(cls, svc_config, running_daemons):
+        try:
+            return next(config for config in svc_config
+                        if config['daemon_name'] in running_daemons)
+        except StopIteration:
+            return None
+
+    @classmethod
+    def _get_default_service(cls, gateways):
+        if gateways:
+            service_name = list(gateways.keys())[0]
+            return service_name, gateways[service_name][0]['service_url']
+        return None
```

Comment:
None of these `@classmethods` use the `cls` argument. In pylint [there's a checker that complains if a instance method doesn't use the `self`](https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/no-self-use.html) at all, but surprisingly there's nothing like that for `@classmethod`s not using `cls`.

Nothing serious, but to me, `staticmethods` are usually a smell that the methods should be attached to some other class instead, like the `list_daemons()` with the `status` attribute.

Reply:
ah shoot, i just went with the force of habit and created them.. i'll sent a PR to fix that. thanks for catching it.

Reply:
i am trying to address this through [this commit.](https://github.com/ceph/ceph/pull/59870/commits/b534f60ed5f415eb5757f734e1c0c6a292c631e9). I have removed the classmethod decorators but I kept the functions inside the same file though. Its mostly fetching things for the configuration and I don't know a separate class for it makes sense..